test/index.js

```diff
@@ -195,31 +200,56 @@ describe('global-proxy', function() {
         var expectConnect = 'example.dev:' + (innerSecure ? 443 : 80);
         var whichAgent = innerSecure ? https.globalAgent : http.globalAgent;
 
-        sinon.assert.calledOnce(whichAgent.request);
-        sinon.assert.calledWith(whichAgent.request, sinon.match.has('method', 'CONNECT'));
-        sinon.assert.calledWith(
-          whichAgent.request,
-          sinon.match.has('path', expectConnect)
-        );
+        if(testParams.proxyEnableFunction !== false) {
```

Comment:
Tests check for the existence of the option, not for its return value depending on the parameters it received